src/fu-engine.c

```diff
@@ -1216,6 +1224,76 @@ fu_engine_composite_cleanup (FuEngine *self, GPtrArray *devices, GError **error)
 	return TRUE;
 }
 
+static GPtrArray *
+fu_engine_get_replugged_devices (FuEngine *self, GPtrArray *devices_old)
+{
+	GPtrArray *devices_new;
+
+	devices_new = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
+	for (guint i = 0; i < devices_old->len; i++) {
+		FuDevice *device;
+		g_autoptr(FuDevice) device_new = NULL;
+		g_autoptr(GError) error_local = NULL;
+		device = g_ptr_array_index (devices_old, i);
+		device_new = fu_device_list_get_by_id (self->device_list,
+						       fu_device_get_id (device),
+						       &error_local);
+		if (device_new == NULL) {
+			g_debug ("failed to find new device: %s",
+				 error_local->message);
+			continue;
+		}
+		g_ptr_array_add (devices_new, g_steal_pointer (&device_new));
+	}
+
+	return devices_new;
+}
+
+gboolean
+fu_engine_install_blob_with_composite (FuEngine *self,
+				       FuDevice *device_orig,
+				       GBytes *blob_cab,
+				       GBytes *blob_fw2,
+				       const gchar *version,
+				       FwupdInstallFlags flags,
+				       GError **error)
+{
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autoptr(GPtrArray) devices_new = NULL;
+
+	devices = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
+	g_ptr_array_add (devices, g_object_ref (device_orig));
```

Comment:
Only the device itself? I guess that's all we've got...

Reply:
Yeah it should just be the one device since it's being called with `install-blob` and you pick one argument.